File: src/main/java/io/bloco/faker/components/Name.java
package io.bloco.faker.components;

import java.util.List;

import io.bloco.faker.FakerComponent;
import io.bloco.faker.FakerData;

public class Name extends FakerComponent {

    public Name(FakerData data) {
        super(data);
    }

    public String firstName() {
        return fetch("name.first_name");
    }

    public String lastName() {
        return fetch("name.last_name");
    }

    public String prefix() {
        return fetch("name.prefix");
    }

    public String suffix() {
        return fetch("name.suffix");
    }

    public String title() {
        return fetch("name.title.descriptor")
                + " " + fetch("name.title.level")
                + " " + fetch("name.title.job");
    }

    public String name() {
        return parse(fetch("name.name"));
    }

    public String nameWithMiddle() {
        return parse(fetch("name.name_with_middle"));
    }

    public List<String> jobTitles() {
        return (List<String>) getMap("name", "title").get("job");
    }
}

File: lib/src/main/kotlin/io/bloco/faker/components/Name.kt
package io.bloco.faker.components

import io.bloco.faker.FakerComponent
import io.bloco.faker.FakerData

class Name(data: FakerData) : FakerComponent(data) {
    fun firstName(): String {
        return fetch("name.first_name")
    }

    fun lastName(): String {
        return fetch("name.last_name")
    }

    fun prefix(): String {
        return fetch("name.prefix")
    }

    fun suffix(): String {
        return fetch("name.suffix")
    }

    fun title(): String {
        return (fetch("name.title.descriptor")
                + " " + fetch("name.title.level")
                + " " + fetch("name.title.job"))
    }

    fun name(): String {
        return parse(fetch("name.name"))
    }

    fun nameWithMiddle(): String {
        return parse(fetch("name.name_with_middle"))
    }

    fun jobTitles(): List<String> {
        return getMap("name", "title")["job"] as List<String>
    }
}